File: ProyectoMysql/server/DataLayer/MerListaDB.py
from EntityLayer.MerListaEntity import MerListaItemModel, MerListaMainModel, MerListaSaveModel, MerListaTituloModel
from Utilidades.Entidades.ResponseAPI import ResponseAPI
from Utilidades.Conexion.ErrorData import ErrorData
from Utilidades.Conexion.configMysql import DBProcedure, Restore, CloseConnection
from Utilidades.Enumerado.ProcessActionEnum import ProcessActionEnum


class MerListaDB:
    def GetItems(Codigo: any):
        try:
            args = (Codigo,)
            resulset = DBProcedure().DBProcedureConsult("sp_MerListaObtenerMain", args)
            list = [MerListaMainModel.Cargar(row) for row in resulset]
            return list
        except Exception as e:
            print(e)

    def Save(Ent: MerListaSaveModel):
        try:
            store_mapping = {
                ProcessActionEnum.Update: "sp_MerLista_Actualizar",
                ProcessActionEnum.Add: "sp_MerLista_Registrar",
            }
            Store = store_mapping.get(Ent.Action, "sp_MerLista_Registrar")
            args = []
            args.append(Ent.ListaId)
            args.append(Ent.CampoId)
            args.append(Ent.Codigo)
            args.append(Ent.Nombre)
            args.append(Ent.Descripcion)
            args.append(Ent.FechaRegistro)
            args.append(Ent.CodUsuario)
            args.append(Ent.EstadoRegistro)
            args.append(Ent.CodigoTabla)
            Ent.ListaId = DBProcedure().DBProcedureInsertUpdate(
                Store, args, "v_ListaId"
            )

            return Ent
        except Exception as e:
            print(e)
            Restore()
            
    def GetItemTitulo(Codigo: any):
        try:
            args = (Codigo,)
            resulset = DBProcedure().DBProcedureConsult("sp_MerListaObtenertitulo", args)
            list = [MerListaTituloModel.CargarItem(row) for row in resulset]
            return list
        except Exception as e:
            print(e)

    def GetItemLike(Codigo: str, Nombre: str):
        try:
            args = (
                Codigo,
                Nombre,
            )
            resulset = DBProcedure().DBProcedureConsult(
                "sp_MerListaBuscarItem", args
            )
            list = [MerListaItemModel.Cargar(row) for row in resulset]
            return list
        except Exception as e:
            print(e)

    def GetItem(Id: int):
        try:
            args = (Id,)
            resulset = DBProcedure().DBProcedureConsult("sp_MerListaObtenerItem", args)
            list = [MerListaItemModel.Cargar(row) for row in resulset]
            return list
        except Exception as e:
            print(e)

File: ProyectoMysql/Api/LogisticStorage/LogisticStorage.DataLayer/MerListaDB.cs
using Framework;
using Framework.Data;
using LogisticStorage.Common;
using LogisticStorage.EntityLayer;
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LogisticStorage.DataLayer
{
    public class MerListaDB : BaseDataEntity
    {
        public virtual List<MerListaEntity> ObtenerMain()
        {
            try
            {
                StartHelper(false);
                IDataReader dr = (IDataReader)DbDatabase.ExecuteReader(System.Data.CommandType.StoredProcedure, "sp_MerListaObtenerMain");
                FillSchemeTable(dr);
                List<MerListaEntity> EntityList = new List<MerListaEntity>();

                while (dr.Read())
                {
                    MerListaEntity entity = new MerListaEntity();
                    if (FillFrom<MerListaEntity>(dr, entity)) EntityList.Add(entity);
                    entity.OnLogicalLoaded();
                }

                Helper.Close(dr);
                return EntityList;
            }
            catch (Exception ex)
            {
                throw ex;
            }
        }

        public virtual bool Registrar(MerListaEntity Ent)
        {
            StartHelper(true);
            try
            {
                RegistrarDB(Ent);
            }
            catch (Exception ex)
            {
                Helper.CancelTransaction();
                throw ex;
            }

            Helper.Close();
            return true;
        }

        private bool RegistrarDB(MerListaEntity Ent)
        {
            if (Ent.LogicalState == LogicalState.Added || Ent.LogicalState == LogicalState.Updated)
            {
                String storedName = "sp_MerLista_Actualizar";
                if (Ent.LogicalState == LogicalState.Added) storedName = "sp_MerLista_Registrar";
                DbDatabase.GetStoredProcCommand(storedName);
                DbDatabase.SetTransaction(Helper.DbTransaction);

                DbDatabase.AddParameter(MyUtils.GetOutputDirection(true), "v_ListaId", DbType.Int32, 4, false, 0, 0, Ent.ListaId);
                DbDatabase.AddParameter(MyUtils.GetOutputDirection(false), "v_CampoId", DbType.Int32, 4, false, 0, 0, Ent.CampoId);
                DbDatabase.AddParameter(MyUtils.GetOutputDirection(false), "v_Codigo", DbType.String, 100, false, 0, 0, Ent.Codigo);
                DbDatabase.AddParameter(MyUtils.GetOutputDirection(false), "v_Nombre", DbType.String, 100, false, 0, 0, Ent.Nombre);
                DbDatabase.AddParameter(MyUtils.GetOutputDirection(false), "v_Descripcion", DbType.String, 100, false, 0, 0, Ent.Descripcion);
                DbDatabase.AddParameter(MyUtils.GetOutputDirection(false), "v_CodUsuario", DbType.String, 20, false, 0, 0, Ent.CodUsuario);
                DbDatabase.AddParameter(MyUtils.GetOutputDirection(false), "v_FechaRegistro", DbType.DateTime, 12, false, 0, 0, Ent.FechaRegistro);
                DbDatabase.AddParameter(MyUtils.GetOutputDirection(false), "v_EstadoRegistro", DbType.Boolean, 2, false, 0, 0, Ent.EstadoRegistro);
                DbDatabase.AddParameter(MyUtils.GetOutputDirection(false), "v_CodigoTabla", DbType.String, 100, false, 0, 0, Ent.CodigoTabla);
                int returnValue = DbDatabase.ExecuteNonQuery();
                if (Ent.LogicalState == LogicalState.Added)
                {
                    if (Ent.ListaId <= 0) Ent.ListaId = (Int32)DbDatabase.GetParameterValue("v_ListaId");
                    Ent.OnLogicalAdded();
                }
                else
                {
                    if (returnValue <= 0) throw new Exception("ErrorDB.UpdateEntity");
                    Ent.OnLogicalUpdate();
                }
            }


            return true;
        }

        public virtual List<MerListaEntity> ObtenerTitulo(String Codigo)
        {
            try
            {
                StartHelper(false);
                DbDatabase.AddParameter(MyUtils.GetOutputDirection(false), "v_Codigo", DbType.String, 50, false, 0, 0, Codigo);
                IDataReader dr = (IDataReader)DbDatabase.ExecuteReader(CommandType.StoredProcedure, "sp_MerListaObtenertitulo");
                FillSchemeTable(dr);
                List<MerListaEntity> EntityList = new List<MerListaEntity>();
                while (dr.Read())
                {
                    MerListaEntity entity = new MerListaEntity();
                    if (FillFrom(dr, entity)) EntityList.Add(entity);
                    entity.OnLogicalLoaded();
                }

                Helper.Close(dr);
                return EntityList;
            }
            catch (Exception ex)
            {
                throw ex;
            }
        }

        public virtual List<MerListaEntity> BuscarItem(String Codigo, String Nombre)
        {
            try
            {
                StartHelper(false);
                DbDatabase.AddParameter(MyUtils.GetOutputDirection(false), "v_Codigo", DbType.String, 50, false, 0, 0, Codigo);
                DbDatabase.AddParameter(MyUtils.GetOutputDirection(false), "v_Nombre", DbType.String, 50, false, 0, 0, Nombre);
                IDataReader dr = (IDataReader)DbDatabase.ExecuteReader(CommandType.StoredProcedure, "sp_MerListaBuscarItem");
                FillSchemeTable(dr);
                List<MerListaEntity> EntityList = new List<MerListaEntity>();
                while (dr.Read())
                {
                    MerListaEntity entity = new MerListaEntity();
                    if (FillFrom(dr, entity)) EntityList.Add(entity);
                    entity.OnLogicalLoaded();
                }

                Helper.Close(dr);
                return EntityList;
            }
            catch (Exception ex)
            {
                throw ex;
            }
        }

        public virtual List<MerListaEntity> ObtenerItem(Int32 MercaderiaId)
        {
            try
            {
                StartHelper(false);
                DbDatabase.AddParameter(MyUtils.GetOutputDirection(false), "v_ListaId", DbType.Int32, 4, false, 0, 0, MercaderiaId);
                IDataReader dr = (IDataReader)DbDatabase.ExecuteReader(CommandType.StoredProcedure, "sp_MerListaObtenerItem");
                FillSchemeTable(dr);
                List<MerListaEntity> EntityList = new List<MerListaEntity>();
                while (dr.Read())
                {
                    MerListaEntity entity = new MerListaEntity();
                    if (FillFrom(dr, entity)) EntityList.Add(entity);
                    entity.OnLogicalLoaded();
                }

                Helper.Close(dr);
                return EntityList;
            }
            catch (Exception ex)
            {
                throw ex;
            }
        }
    }
}
